tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -117,6 +123,12 @@ sealed trait NativeConfig {
   /** Create a new config with the given lto mode. */
   def withLTO(value: LTO): NativeConfig
 
+  /** Create a new config with the given max inline depth. */
+  def withMaxInlineDepth(value: Int): NativeConfig
```

Comment:
In `Inline` we define also a max size of inlined methods. Currently, it's hard coded to 8 instructions. Maybe we can also allow to tune this condition as well